hack/run-locally.sh

```diff
@@ -31,13 +31,40 @@ function override_install_manifests() {
     fi
 }
 
-# Extract environment variables for the CNO DaemonSet
+# Extract and update CNO environment variables
 function setup_operator_env() {
-    if [[ manifests/0000_70_cluster-network-operator_03_daemonset.yaml -nt "${CLUSTER_DIR}/env.sh" ]]; then
-        echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
-        oc --kubeconfig=hack/null-kubeconfig patch --local=true -f manifests/0000_70_cluster-network-operator_03_daemonset.yaml -p '{}' -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+    local mode="$1"
+    local image_env_key="$2"
+    local plugin_image="$3"
+
+    if [[ "${mode}" == "attach" ]]; then
+        # Extract environment from a running cluster
+        if [[ ! -e "${CLUSTER_DIR}/env.sh" ]]; then
+            echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
+            kubectl get deployment -n openshift-network-operator network-operator -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+        fi
+    else
+        # Extract environment from the CNO's manifests
+        if [[ manifests/0000_70_cluster-network-operator_03_daemonset.yaml -nt "${CLUSTER_DIR}/env.sh" ]]; then
+            echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
+            oc --kubeconfig=hack/null-kubeconfig patch --local=true -f manifests/0000_70_cluster-network-operator_03_daemonset.yaml -p '{}' -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+        fi
     fi
+
     sed -i -e "s/^RELEASE_VERSION=.*/RELEASE_VERSION=${RELEASE_VERSION}/" "${CLUSTER_DIR}/env.sh"
+
+    if [[ -n "${plugin_image}" ]]; then
+        sed -i -e "s#^${image_env_key}=.*#${image_env_key}=${plugin_image}#" "${CLUSTER_DIR}/env.sh"
+    fi
+}
+
+# Install our overrides in to an existing cluster
+function stop_deployed_operator() {
```

Comment:
I initially wrote "comment does not match function", but I guess it does, except that the comment and the function name describe what the function is doing from completely different perspectives, so they seem mismatched